packages/web-components/src/components/image-with-caption/image-with-caption.ts

```diff
@@ -0,0 +1,100 @@
+/**
+ * @license
+ *
+ * Copyright IBM Corp. 2020
+ *
+ * This source code is licensed under the Apache-2.0 license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+import { html, property, customElement, LitElement } from 'lit-element';
+import settings from 'carbon-components/es/globals/js/settings';
+import ddsSettings from '@carbon/ibmdotcom-utilities/es/utilities/settings/settings.js';
+import '../image/image';
+import '../lightbox-media-viewer/lightbox-image-viewer';
+import '../button/button';
+import ZoomIn20 from 'carbon-web-components/es/icons/zoom--in/20';
+import DDSModal from '../modal/modal';
+import 'carbon-web-components/es/components/modal/modal-close-button';
+import styles from './image-with-caption.scss';
+
+const { prefix } = settings;
+const { stablePrefix: ddsPrefix } = ddsSettings;
+
+/**
+ * Image With Caption
+ *
+ * @element dds-image-with-caption
+ */
+
+@customElement(`${ddsPrefix}-image-with-caption`)
+class DDSImageWithCaption extends LitElement {
+  /**
+   * The lightbox.
+   */
+  @property({ type: Boolean, reflect: true })
+  lightbox = false;
+
+  /**
+   * The image source.
+   */
+  @property({ reflect: true })
+  defaultSrc = '';
+
+  /**
+   * The alt text.
+   */
+  @property({ reflect: true })
+  alt = '';
+
+  /**
+   * The heading.
+   */
+  @property({ reflect: true })
+  heading = '';
+
+  /**
+   * The description.
+   */
+  @property({ reflect: true })
+  copy = '';
+
+  _handleClick() {
+    const modal = this.shadowRoot?.querySelector('dds-modal');
+    (modal as DDSModal).open = true;
+  }
+
+  render() {
+    return html`
+      ${this.lightbox
+        ? html`
+            <button class="${prefix}--image-with-caption__image" @click="${this._handleClick}">
+              <dds-image alt="${this.alt}" default-src="${this.defaultSrc}" />
+              <div class="${prefix}--image-with-caption__zoom-button">
+                ${ZoomIn20()}
+              </div>
+            </button>
+            <dds-modal expressive-size="full-width">
+              <bx-modal-close-button></bx-modal-close-button>
+              <dds-lightbox-image-viewer
+                alt="${this.alt}"
+                default-src="${this.defaultSrc}"
+                description="${this.copy}"
+                title="${this.heading}"
+              >
+              </dds-lightbox-image-viewer>
+            </dds-modal>
```

Comment:
If a component renders a modal, it has to be at a separate DOM tree from where the component is at, to ensure the modal covers up the entire body content when it shows up. Doing so probably eliminates the need for [`:host(#{$dds-prefix}-image-with-caption) #{$dds-prefix}-modal { left: 0; }` style](https://github.com/carbon-design-system/ibm-dotcom-library/pull/4065/files#r495361740).

We can use [`ModalRenderMixin`](https://github.com/carbon-design-system/ibm-dotcom-library/blob/v1.11.0/packages/web-components/src/globals/mixins/modal-render.ts) for that; With it, putting rendering the modal content via `renderModal()` ensures that the modal is put at the right place ([example](https://github.com/carbon-design-system/ibm-dotcom-library/blob/v1.11.0/packages/web-components/src/components/lightbox-media-viewer/lightbox-video-player-composite.ts#L89-L107)).

Reply:
@asudoh Ah makes sense! Just updated the PR.